options/logger.go

```diff
@@ -77,95 +77,120 @@ func (f RawLoggerConfigFormat) Validate() error {
 	}
 }
 
-func (f RawLoggerConfigFormat) unmarshal(data []byte, out LoggerProducer) error {
+// Unmarshal unmarshals the given data using the corresponding unmarshaler for
+// the RawLoggerConfigFormat type.
+func (f RawLoggerConfigFormat) Unmarshal(data []byte, out interface{}) error {
 	switch f {
 	case RawLoggerConfigFormatBSON:
 		if err := bson.Unmarshal(data, out); err != nil {
 			return errors.Wrapf(err, "could not render '%s' input into '%s'", data, out)
 
 		}
-
-		return nil
 	case RawLoggerConfigFormatJSON:
 		if err := json.Unmarshal(data, out); err != nil {
 			return errors.Wrapf(err, "could not render '%s' input into '%s'", data, out)
 
 		}
-
-		return nil
 	default:
 		return errors.Errorf("unsupported format '%s'", f)
 	}
+
+	return nil
+}
+
+// RawLoggerConfig wraps []byte and implements the json and bson Marshaler and
+// Unmarshaler interfaces.
+type RawLoggerConfig []byte
+
+func (lc *RawLoggerConfig) MarshalJSON() ([]byte, error) {
+	if len(*lc) > 0 {
+		return *lc, nil
+	}
+
+	return json.Marshal([]byte(*lc))
+}
+
+func (lc *RawLoggerConfig) UnmarshalJSON(b []byte) error {
```

Comment:
do you need to have unmarshal functions? I think it should just work...

Reply:
no it fails, i originally didn't have them and it complains that it can't unmarshal